torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -199,7 +199,8 @@ def runtime_wrapper(args: List[Any]):
             for idx in indices_of_inps_to_detach:
                 if isinstance(args_[idx], torch.Tensor):
                     args_[idx] = args_[idx].detach()
-            with torch.autograd._force_original_view_tracking(True):
+            needs_grad = any(x.requires_grad for x in args_ if isinstance(x, Tensor))
```

Comment:
I don't think we want to determine `need_grad` here by looking at if any of the inputs require grad. Instead, you can just assume that if `trace_joint == True`, we should be turning on `enable_grad` (because trace_joint implies that we need a training graph, and we need at least one output that requires grad.

Can you also add a comment explaining why we need the enable_grad at runtime? Namely that it's possible for a user to run a compiled region under no_grad, but still require some of the outputs of the graph to require gradients.